From 4814b3b49989ff5e03abaac4c9e03b12412f665d Mon Sep 17 00:00:00 2001
From: Robby Cai <r63905@freescale.com>
Date: Tue, 10 Jun 2008 17:38:40 +0800
Subject: [PATCH 022/442] ENGR00079160 VPU lib v3.0.0 not work on MX27 TO2.1

Fix: VPU lib v3.0.0 does not work on MX27 TO2.1, but work on TO2.0.
And some restruction

Signed-off-by: Robby Cai <r63905@freescale.com>
---
 test/mxc_vpu_test/dec.c     | 20 ++++++++++----------
 test/mxc_vpu_test/display.c |  6 +++---
 test/mxc_vpu_test/enc.c     |  4 ++--
 test/mxc_vpu_test/fb.c      |  4 ++--
 4 files changed, 17 insertions(+), 17 deletions(-)

diff --git a/test/mxc_vpu_test/dec.c b/test/mxc_vpu_test/dec.c
index 065080b..61fd7b1 100644
--- a/test/mxc_vpu_test/dec.c
+++ b/test/mxc_vpu_test/dec.c
@@ -415,7 +415,7 @@ decoder_start(struct decode *dec)
 					pfb->desc.phy_addr;
 		
 			
-			if (platform_is_mxc30031()) {
+			if (cpu_is_mxc30031()) {
 				write_to_file(dec, (u8 *)yuv_addr);
 			} else {
 				fwriten(dec->cmdl->dst_fd, (u8 *)yuv_addr,
@@ -469,7 +469,7 @@ decoder_free_framebuffer(struct decode *dec)
 		v4l_display_close(dec->disp);
 
 		if ((rot_en == 0) && (mp4dblk_en == 0) && (dering_en == 0)) {
-			if (platform_is_mx37()) {
+			if (cpu_is_mx37()) {
 				for (i = 0; i < dec->fbcount; i++) {
 					IOFreePhyMem(&mvcol_md[i]);
 				}
@@ -481,7 +481,7 @@ decoder_free_framebuffer(struct decode *dec)
 	if ((dec->cmdl->dst_scheme != PATH_V4L2) ||
 			((dec->cmdl->dst_scheme == PATH_V4L2) &&
 			 (dec->cmdl->rot_en || dec->cmdl->mp4dblk_en ||
-			 (platform_is_mx37() && dec->cmdl->dering_en)))) {
+			 (cpu_is_mx37() && dec->cmdl->dering_en)))) {
 		totalfb = dec->fbcount + dec->extrafb;
 		for (i = 0; i < totalfb; i++) {
 			framebuf_free(dec->pfbpool[i]);
@@ -539,7 +539,7 @@ decoder_allocate_framebuffer(struct decode *dec)
 	
 	if ((dst_scheme != PATH_V4L2) ||
 		((dst_scheme == PATH_V4L2) && (rot_en || mp4dblk_en ||
-			(platform_is_mx37() && dering_en)))) {
+			(cpu_is_mx37() && dering_en)))) {
 
 		for (i = 0; i < totalfb; i++) {
 			pfbpool[i] = framebuf_alloc(dec->stride,
@@ -552,7 +552,7 @@ decoder_allocate_framebuffer(struct decode *dec)
 			fb[i].bufY = pfbpool[i]->addrY;
 			fb[i].bufCb = pfbpool[i]->addrCb;
 			fb[i].bufCr = pfbpool[i]->addrCr;
-			if (platform_is_mx37()) {
+			if (cpu_is_mx37()) {
 				fb[i].bufMvCol = pfbpool[i]->mvColBuf;
 			}
 		}
@@ -562,7 +562,7 @@ decoder_allocate_framebuffer(struct decode *dec)
 		rotation.rot_en = dec->cmdl->rot_en;
 		rotation.rot_angle = dec->cmdl->rot_angle;
 		/* use the ipu h/w rotation instead of vpu rotaton */
-		if (platform_is_mx37() && rotation.rot_en) {
+		if (cpu_is_mx37() && rotation.rot_en) {
 			rotation.rot_en = 0;
 			rotation.ipu_rot_en = 1;
 		}
@@ -576,7 +576,7 @@ decoder_allocate_framebuffer(struct decode *dec)
 		if ((rot_en == 0) && (mp4dblk_en == 0) && (dering_en == 0)) {
 			img_size = dec->stride * dec->picheight;
 
-			if (platform_is_mx37()) {
+			if (cpu_is_mx37()) {
 				mvcol_md = dec->mvcol_memdesc =
 					calloc(fbcount, sizeof(vpu_mem_desc));
 			}
@@ -585,7 +585,7 @@ decoder_allocate_framebuffer(struct decode *dec)
 				fb[i].bufCb = fb[i].bufY + img_size;
 				fb[i].bufCr = fb[i].bufCb + (img_size >> 2);
 				/* allocate MvCol buffer here */
-				if (platform_is_mx37()) {
+				if (cpu_is_mx37()) {
 					memset(&mvcol_md[i], 0,
 							sizeof(vpu_mem_desc));
 					mvcol_md[i].size = img_size >> 2;
@@ -623,7 +623,7 @@ err1:
 err:
 	if ((dst_scheme != PATH_V4L2) ||
 		((dst_scheme == PATH_V4L2) && (rot_en || mp4dblk_en ||
-			(platform_is_mx37() && dering_en)))) {
+			(cpu_is_mx37() && dering_en)))) {
 		for (i = 0; i < totalfb; i++) {
 			framebuf_free(pfbpool[i]);
 		}
@@ -637,7 +637,7 @@ err:
 static void
 calculate_stride(struct decode *dec)
 {
-	if (platform_is_mxc30031()) {
+	if (cpu_is_mxc30031()) {
 		if (dec->picwidth <= 128) {
 			dec->stride = 128;
 		} else if (dec->picwidth <= 256) {
diff --git a/test/mxc_vpu_test/display.c b/test/mxc_vpu_test/display.c
index 9a3bf01..8c766c6 100644
--- a/test/mxc_vpu_test/display.c
+++ b/test/mxc_vpu_test/display.c
@@ -84,7 +84,7 @@ v4l_display_open(int width, int height, int nframes, struct rot rotation,
 	struct vpu_display *disp;
 	int ratio = 1;
 	
-	if (platform_is_mx27()) {
+	if (cpu_is_mx27()) {
 		out = 0;
 	} else {
 		out = 3;
@@ -148,7 +148,7 @@ v4l_display_open(int width, int height, int nframes, struct rot rotation,
 	crop.c.height = height / ratio;
 	dprintf(1, "crop.c.width/height: %d/%d\n", width/ratio, height/ratio);
 	
-	if (platform_is_mx37()) {
+	if (cpu_is_mx37()) {
 		crop.c.width = cropcap.bounds.width;
 		crop.c.height = cropcap.bounds.height;
 	}
@@ -183,7 +183,7 @@ v4l_display_open(int width, int height, int nframes, struct rot rotation,
 		}
 	}
 
-	if (platform_is_mx27()) {
+	if (cpu_is_mx27()) {
 		fb.capability = V4L2_FBUF_CAP_EXTERNOVERLAY;
 		fb.flags = V4L2_FBUF_FLAG_PRIMARY;
 	
diff --git a/test/mxc_vpu_test/enc.c b/test/mxc_vpu_test/enc.c
index bdc9f95..85488b8 100644
--- a/test/mxc_vpu_test/enc.c
+++ b/test/mxc_vpu_test/enc.c
@@ -409,9 +409,9 @@ encoder_configure(struct encode *enc)
 	RetCode ret;
 	MirrorDirection mirror;
 
-	if (platform_is_mx27()) {
+	if (cpu_is_mx27()) {
 		search_pa.searchRamAddr = 0xFFFF4C00;
-	} else if (platform_is_mxc30031()) {
+	} else if (cpu_is_mxc30031()) {
 		search_pa.searchRamAddr = 0xD0000000;
 		search_pa.SearchRamSize =
 			((enc->picwidth + 15) & ~15) * 36 + 2048;
diff --git a/test/mxc_vpu_test/fb.c b/test/mxc_vpu_test/fb.c
index 5c46c5f..81c8757 100644
--- a/test/mxc_vpu_test/fb.c
+++ b/test/mxc_vpu_test/fb.c
@@ -74,7 +74,7 @@ struct frame_buf *framebuf_alloc(int strideY, int height)
 
 	memset(&(fb->desc), 0, sizeof(vpu_mem_desc));
 	fb->desc.size = (strideY * height * 3 / 2);
-	if (platform_is_mx37())
+	if (cpu_is_mx37())
 		fb->desc.size += strideY * height / 4;
 	
 	err = IOGetPhyMem(&fb->desc);
@@ -87,7 +87,7 @@ struct frame_buf *framebuf_alloc(int strideY, int height)
 	fb->addrY = fb->desc.phy_addr;
 	fb->addrCb = fb->addrY + strideY * height;
 	fb->addrCr = fb->addrCb + strideY / 2 * height / 2;
-	if (platform_is_mx37()) {
+	if (cpu_is_mx37()) {
 		fb->mvColBuf = fb->addrCr + strideY / 2 * height / 2;
 	}
 	fb->desc.virt_uaddr = IOGetVirtMem(&(fb->desc));
-- 
1.8.0

